src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseHelper.CoreClr.cs

```diff
@@ -35,12 +35,31 @@ internal static class WebResponseHelper
             return headers;
         }
 
-        internal static string GetOutFilePath(HttpResponseMessage response, string _qualifiedOutFile)
-        {
-            // Get file name from last segment of Uri
-            string lastUriSegment = System.Net.WebUtility.UrlDecode(response.RequestMessage.RequestUri.Segments[^1]);
+        internal static string GetOutFilePath(HttpResponseMessage response, string qualifiedOutFile)
```

Comment:
We should be more smart - check File.Exists for result file name and if there is a conflict add `(1)` or `(2)` and so on.

Reply:
curl doesn't check for existing files and just overwrites